cpp/src/io/utilities/wrapper_utils.hpp

```diff
@@ -146,3 +146,70 @@ struct rmm_deleter {
 };
 template <typename T>
 using device_ptr = std::unique_ptr<T, rmm_deleter<T>>;
+
+/**
+ * @brief A helper class that owns a resizable device memory buffer.
```

Comment:
I'd make it clear in the documentation that the buffer is uninitialized.

Reply:
@jrhemstad added the info below. 
The PR should be ready for merge now. Please update your review if you don't have any other suggestions.